modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,424 @@
-            elog("Error running against host #{tip}: #{e.message}\n#{e.backtrace.join("\n")}")
-          end
-        end
-      end
-      # Exit once we run out of hosts
-      if(tl.length == 0)
-        break
-      end
-      tl.first.join
-      tl.delete_if { |t| not t.alive? }
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::A
+      records << "#{r.address}"
+      report_host(host: r.address, name: domain, info: 'A')
+      print_good("#{domain} A: #{r.address} ") if datastore['ENUM_BRT']
     end
+    return if records.blank?
+    records
   end
 
-  # SRV Record Enumeration
-  def srvqry(dom,nssrv)
-    print_status("Enumerating SRV records for #{dom}")
-    i, a = 0, []
-    # Most common SRV Records
-    srvrcd = [
-      "_gc._tcp.","_kerberos._tcp.", "_kerberos._udp.","_ldap._tcp.","_test._tcp.",
-      "_sips._tcp.","_sip._udp.","_sip._tcp.","_aix._tcp.","_aix._tcp.","_finger._tcp.",
-      "_ftp._tcp.","_http._tcp.","_nntp._tcp.","_telnet._tcp.","_whois._tcp.","_h323cs._tcp.",
-      "_h323cs._udp.","_h323be._tcp.","_h323be._udp.","_h323ls._tcp.","_h323ls._udp.",
-      "_sipinternal._tcp.","_sipinternaltls._tcp.","_sip._tls.","_sipfederationtls._tcp.",
-      "_jabber._tcp.","_xmpp-server._tcp.","_xmpp-client._tcp.","_imap._tcp.","_certificates._tcp.",
-      "_crls._tcp.","_pgpkeys._tcp.","_pgprevokations._tcp.","_cmp._tcp.","_svcp._tcp.","_crl._tcp.",
-      "_ocsp._tcp.","_PKIXREP._tcp.","_smtp._tcp.","_hkp._tcp.","_hkps._tcp.","_jabber._udp.",
-      "_xmpp-server._udp.","_xmpp-client._udp.","_jabber-client._tcp.","_jabber-client._udp."]
-    srvrcd.each do |srvt|
-      trg = "#{srvt}#{dom}"
-      query = @res.query(trg , Net::DNS::SRV)
-      next unless query
-      query.answer.each do |srv|
-        next if srv.type == "CNAME"
-        print_status("SRV Record: #{trg} Host: #{srv.host} Port: #{srv.port} Priority: #{srv.priority}")
-      end
+  def get_cname(domain)
+    print_status("querying DNS CNAME records for #{domain}")
+    resp = dns_query(domain, 'CNAME')
+    return if resp.blank? || resp.answer.blank?
+
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::CNAME
+      records << r.cname
+      print_good("#{domain} CNAME: #{r.cname}")
     end
-    print_status("Done")
+    return if records.blank?
+    records
   end
 
-  # For Performing Zone Transfers
-  def axfr(target, nssrv)
-    print_status("Performing zone transfer against all nameservers in #{target}")
-    if not nssrv.nil?
-      @res.nameserver=(nssrv)
-      @nsinuse = nssrv
+  def get_ns(domain)
+    print_status("querying DNS NS records for #{domain}")
+    resp = dns_query(domain, 'NS')
+    return if resp.blank? || resp.answer.blank?
+
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::NS
+      records << "#{r.nsdname}"
+      report_host(host: r.nsdname, name: domain, info: 'NS')
```

Comment:
IMO, this should be removed.  While we do correctly get the `NS` records:

```
[+] rapid7.com NS: blah.rapid7.com.
```

But the corresponding entry in `hosts` resolves the discovered nameserver's IP and stores that instead.  So, good idea, but I'm not sure there is a good way to store this today with `report_host`.  `report_note` is probably a better option.  This goes for all record types that you are calling `report_host` on after the fact.